tasks/Functions-billing/fund.js

```diff
@@ -8,60 +11,32 @@ task("functions-sub-fund", "Funds a billing subscription for Functions consumer
       throw Error("This command cannot be used on a local hardhat chain.  Specify a valid network.")
     }
 
-    const subscriptionId = taskArgs.subid
-    const linkAmount = taskArgs.amount
-
-    const RegistryFactory = await ethers.getContractFactory(
-      "contracts/dev/functions/FunctionsBillingRegistry.sol:FunctionsBillingRegistry"
-    )
-    const registry = await RegistryFactory.attach(networks[network.name]["functionsBillingRegistryProxy"])
-
-    // Check that the subscription is valid
-    let preSubInfo
-    try {
-      preSubInfo = await registry.getSubscription(subscriptionId)
-    } catch (error) {
-      if (error.errorName === "InvalidSubscription") {
-        throw Error(`Subscription ID "${subscriptionId}" is invalid or does not exist`)
-      }
-      throw error
-    }
+    const signer = await ethers.getSigner()
+    const linkTokenAddress = networks[network.name]["linkToken"]
+    const functionsRouterAddress = networks[network.name]["functionsRouter"]
+    const txOptions = { confirmations: networks[network.name].confirmations }
 
-    // Convert LINK to Juels
-    const juelsAmount = ethers.utils.parseUnits(linkAmount)
-    console.log(`Funding subscription ${subscriptionId} with ${ethers.utils.formatEther(juelsAmount)} LINK`)
+    const subId = taskArgs.subid
+    const linkAmount = taskArgs.amount
+    const juelsAmount = ethers.utils.parseUnits(linkAmount, 18)
 
-    const LinkTokenFactory = await ethers.getContractFactory("LinkToken")
-    const linkToken = await LinkTokenFactory.attach(networks[network.name].linkToken)
+    const sm = new SubscriptionManager({ signer, linkTokenAddress, functionsRouterAddress })
+    await sm.initialize()
 
-    const accounts = await ethers.getSigners()
-    const signer = accounts[0]
+    await utils.prompt(
+      `\nPlease confirm that you wish to fund Subscription '${subId}' with '${linkAmount}' LINK from your wallet.'`
+    )
 
-    // Ensure sufficient balance
-    const balance = await linkToken.balanceOf(signer.address)
-    if (juelsAmount.gt(balance)) {
-      throw Error(
-        `Insufficient LINK balance. Trying to fund subscription with ${ethers.utils.formatEther(
-          juelsAmount
-        )} LINK, but wallet only has ${ethers.utils.formatEther(balance)}.`
-      )
-    }
+    console.log(`\nFunding subscription '${subId}' with '${linkAmount}' LINK...`)
 
-    // Fund the subscription with LINK
-    const fundTx = await linkToken.transferAndCall(
-      networks[network.name]["functionsBillingRegistryProxy"],
-      juelsAmount,
-      ethers.utils.defaultAbiCoder.encode(["uint64"], [subscriptionId])
-    )
+    const fundTxReceipt = await sm.fundSubscription({ juelsAmount, subId, txOptions })
+    console.log(`\nSubscription '${subId}' funded with '${linkAmount}' LINK in Tx: ''${fundTxReceipt.transactionHash}'`)
 
```

Comment:
Let's create a util function that gets the output from `getSubscriptionInfo` and converts the balance amounts from Juels to LINK, returning the SubInfo object with the converted values.  Then, we can use that util function anywhere we use `getSubscriptionInfo` before displaying to the user.

Reply:
as per slack chat, if we wrap sm.fundSubscription in a helper function we lose educational value on the NPM package as the end user may not dig into the abstraction.  This way we show the sm to the end user directly.  And by showing the conversion being done we show them what the dev needs to do when consuming values returned in juels etc.